tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/field_access.bal

```diff
@@ -353,10 +353,84 @@ function testFieldAccessOnMapConstruct() returns boolean {
     return "Sanjiva" == name;
 }
 
+type FooRec1 record {
+    json j;
+};
+
+type FooRec2 record {
+    json j;
+    FooRec1 fooRec;
+};
+
+function testFieldAccessOnJsonTypedRecordFields() returns error? {
+    FooRec1 rec1 = {j: "1"};
+    string val = check rec1.j;
+    assertEquals(val, "1");
+
+    rec1 = {j: {k: "2"}};
+    val = check rec1.j.k;
+    assertEquals(val, "2");
+
+    rec1 = {j: {k: {l: "1", m: "3"}}};
+    val = check rec1.j.k.m;
+    assertEquals(val, "3");
+
+    FooRec2 rec2 = {j: "1", fooRec: {j: "2"}};
+    val = check rec2.j;
+    assertEquals(val, "1");
+    val = check rec2.fooRec.j;
+    assertEquals(val, "2");
+
+    rec2 = {j: "1", fooRec: {j: {k: "3", l: {m: "4"}}}};
+    val = check rec2.fooRec.j.k;
+    assertEquals(val, "3");
+    val = check rec2.fooRec.j.l.m;
+    assertEquals(val, "4");
+}
+
+function testFieldAccessOnJsonTypedRecordFieldsResultingInErrors() {
+    FooRec1 rec1 = {j: "1"};
+    json|error val = rec1.j.k.l;
```

Comment:
Doesn't the failure happen only with check? We should have the similar scenario here. We can do something like

```ballerina
    var fn = function() returns string|error {
        FooRec1 rec1 = {j: "1"};
        string val = check rec1.j.k.l;
        return val;
    };
    validateKeyNotFoundError(fn());
```

Reply:
It happens without the `check` as well. Will add few test with check as well

Reply:
Updated 8ce9bb79f0adc9cf67ab12d17decf545ba190bdc

Reply:
```ballerina
type FooRec1 record {
    json j;
};

public function main() returns error? {
    FooRec1 rec1 = {j: "1"};
    json|error x = rec1.j.k.l;
}
```

doesn't crash for me on Update 2.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/38328 this is the issue related to bug in 2.x branch, where NPE occurs when access more than 3rd level (NPE for `rec1.j.k.l.p;`).